packages/block-editor/src/components/image-size-control/index.js

```diff
@@ -33,6 +33,15 @@ export default function ImageSizeControl( {
 		};
 	}
 
+	// Normalize the changed dimension before passing it on.
+	function normalizeDimension( dimension ) {
+		const parsedDimension = parseInt( dimension, 10 );
+
+		return isNaN( parsedDimension ) || parsedDimension <= 0
+			? undefined
+			: parsedDimension;
+	}
```

Comment:
Unit tests are pretty cheap, and considering how many times this bug has popped up in a few places I think it'd be worth it. It's also fine later if we decide that it doesn't have much value and we can 🔪 remove it in a follow up PR

So, I won't block the PR on this, but this is pretty easily unit testable if we make a few small changes, with moving the declaration above ImageSizeControl and passing through `onChange` as a parameter. We can then easily mock and spy on the `onChange` function in jest. As a bonus, we only declare these functions once rather than each time during render. (If folks decide to go this route and get stuck with the testing setup I'm happy to provide additional help).

So all together the signature changes might look something like:

```jsx
function normalizeDimension( dimension ) { //can also just export this if you're more interested in testing this
    //...
}

export default function normalizeWidth( value, onChange ) {
  return onChange( { width: normalizeDimension( value ) } );
}

export default function normalizeHeight( value, onChange ) {
 return onChange( { height: normalizeDimension( value ) } );
}

export default function ImageSizeControl( {

//...
<TextControl
	type="number"
	className="block-editor-image-size-control__width"
	label={ __( 'Width' ) }
	value={ width ?? imageWidth ?? '' }
	min={ 1 }
	onChange={ normalizeWidth }
```